Exception at LSPAnnotator. **** guava

**Description:**

Current lsp4intellij throws exception at LSPAnnotator in intellij-language-1c-bsl (with BSL Language Server).

```
ExternalToolPass: 

com.intellij.diagnostic.PluginException: annotator: org.wso2.lsp4intellij.contributors.annotator.LSPAnnotator@13572e99 (class org.wso2.lsp4intellij.contributors.annotator.LSPAnnotator) [Plugin: org.1c-syntax.intellij.language-1c-bsl]
	at com.intellij.ide.plugins.PluginManagerCore.createPluginException(PluginManagerCore.java:403)
	at com.intellij.diagnostic.PluginProblemReporterImpl.createPluginExceptionByClass(PluginProblemReporterImpl.java:12)
	at com.intellij.diagnostic.PluginException.createByClass(PluginException.java:72)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass.process(ExternalToolPass.java:253)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass.doApply(ExternalToolPass.java:219)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass.access$600(ExternalToolPass.java:39)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass$1.lambda$null$0(ExternalToolPass.java:183)
	at com.intellij.openapi.application.ReadAction.lambda$run$1(ReadAction.java:53)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:936)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:57)
	at com.intellij.openapi.application.ReadAction.run(ReadAction.java:53)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass$1.lambda$run$1(ExternalToolPass.java:180)
	at com.intellij.openapi.progress.util.BackgroundTaskUtil.lambda$runUnderDisposeAwareIndicator$11(BackgroundTaskUtil.java:244)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$3(CoreProgressManager.java:185)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:169)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:591)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:537)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:59)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:156)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:185)
	at com.intellij.openapi.progress.util.BackgroundTaskUtil.runUnderDisposeAwareIndicator(BackgroundTaskUtil.java:264)
	at com.intellij.openapi.progress.util.BackgroundTaskUtil.runUnderDisposeAwareIndicator(BackgroundTaskUtil.java:243)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass$1.run(ExternalToolPass.java:178)
	at com.intellij.util.ui.update.MergingUpdateQueue.execute(MergingUpdateQueue.java:324)
	at com.intellij.util.ui.update.MergingUpdateQueue.execute(MergingUpdateQueue.java:314)
	at com.intellij.util.ui.update.MergingUpdateQueue.lambda$flush$1(MergingUpdateQueue.java:264)
	at com.intellij.util.ui.update.MergingUpdateQueue.flush(MergingUpdateQueue.java:278)
	at com.intellij.util.ui.update.MergingUpdateQueue.run(MergingUpdateQueue.java:233)
	at com.intellij.util.concurrency.QueueProcessor.runSafely(QueueProcessor.java:231)
	at com.intellij.util.Alarm$Request.runSafely(Alarm.java:368)
	at com.intellij.util.Alarm$Request.run(Alarm.java:355)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at com.intellij.util.concurrency.SchedulingWrapper$MyScheduledFutureTask.run(SchedulingWrapper.java:224)
	at com.intellij.util.concurrency.BoundedTaskExecutor.doRun(BoundedTaskExecutor.java:207)
	at com.intellij.util.concurrency.BoundedTaskExecutor.access$100(BoundedTaskExecutor.java:29)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.lambda$run$0(BoundedTaskExecutor.java:185)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:208)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.run(BoundedTaskExecutor.java:181)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.IncompatibleClassChangeError: Class com.google.common.base.Predicates$ObjectPredicate$4 does not implement the requested interface java.util.function.Predicate
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:176)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1654)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at org.wso2.lsp4intellij.contributors.annotator.LSPAnnotator.lambda$createAnnotations$1(LSPAnnotator.java:130)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1540)
	at org.wso2.lsp4intellij.contributors.annotator.LSPAnnotator.createAnnotations(LSPAnnotator.java:101)
	at org.wso2.lsp4intellij.contributors.annotator.LSPAnnotator.apply(LSPAnnotator.java:89)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass.doApply(ExternalToolPass.java:216)
	... 37 more
```

**Affected Product Version:**

**OS, DB, other environment details and versions:**    

* windows 10 pro
* IntelliJ IDEA Ultimate 2019.2.1
* AdoptJDK 11
* lsp4intellij a5e62767d0a8c734c305a7bba60bd62bd2aa9273
* intellij-language-1c-bsl [commit 38348a2a61f49af995ce6ceeab1f18864235c6a1](https://github.com/1c-syntax/intellij-language-1c-bsl/commit/38348a2a61f49af995ce6ceeab1f18864235c6a1)

**Steps to reproduce:**

Clone intellij-language-1c-bsl repo, run gradle task `runIDE`

--- a/src/main/java/org/wso2/lsp4intellij/contributors/annotator/LSPAnnotator.java
+++ b/src/main/java/org/wso2/lsp4intellij/contributors/annotator/LSPAnnotator.java
@@ -15,7 +15,6 @@
  */
 package org.wso2.lsp4intellij.contributors.annotator;
 
-import com.google.common.base.Predicates;
 import com.intellij.lang.annotation.Annotation;
 import com.intellij.lang.annotation.AnnotationHolder;
 import com.intellij.lang.annotation.ExternalAnnotator;
@@ -40,6 +39,7 @@ import org.wso2.lsp4intellij.utils.FileUtils;
 
 import java.util.ConcurrentModificationException;
 import java.util.List;
+import java.util.Objects;
 
 public class LSPAnnotator extends ExternalAnnotator {
     private static final Object RESULT = new Object();
@@ -127,7 +127,7 @@ public class LSPAnnotator extends ExternalAnnotator {
             final LSPPsiElement element = new LSPPsiElement(name, editor.getProject(), start, end, file);
             final List<Either<Command, CodeAction>> codeAction = eventManager.codeAction(element);
             if (codeAction != null) {
-                codeAction.stream().filter(Predicates.notNull()).forEach(e -> {
+                codeAction.stream().filter(Objects::nonNull).forEach(e -> {
                     if (e.isLeft()) {
                         annotation.registerFix(new LSPCommandFix(uri, e.getLeft()), textRange);
                     } else if (e.isRight()) {

--- a/src/main/java/org/wso2/lsp4intellij/editor/EditorEventManager.java
+++ b/src/main/java/org/wso2/lsp4intellij/editor/EditorEventManager.java
@@ -15,7 +15,6 @@
  */
 package org.wso2.lsp4intellij.editor;
 
-import com.google.common.base.Strings;
 import com.intellij.codeInsight.completion.InsertionContext;
 import com.intellij.codeInsight.daemon.DaemonCodeAnalyzer;
 import com.intellij.codeInsight.hint.HintManager;
@@ -54,6 +53,7 @@ import com.intellij.psi.PsiDocumentManager;
 import com.intellij.psi.PsiElement;
 import com.intellij.psi.PsiFile;
 import com.intellij.ui.Hint;
+import org.apache.commons.lang3.StringUtils;
 import org.eclipse.lsp4j.CodeAction;
 import org.eclipse.lsp4j.CodeActionContext;
 import org.eclipse.lsp4j.CodeActionParams;
@@ -783,7 +783,7 @@ public class EditorEventManager {
             }
 
             String string = HoverHandler.getHoverString(hover);
-            if (Strings.isNullOrEmpty(string)) {
+            if (StringUtils.isEmpty(string)) {
                 LOG.warn(String.format("Hover string returned is null for file %s and pos (%d;%d)",
                         identifier.getUri(), serverPos.getLine(), serverPos.getCharacter()));
                 return;
@@ -872,7 +872,7 @@ public class EditorEventManager {
         String label = item.getLabel();
         TextEdit textEdit = item.getTextEdit();
         List<TextEdit> addTextEdits = item.getAdditionalTextEdits();
-        String presentableText = !Strings.isNullOrEmpty(label) ? label : (insertText != null) ? insertText : "";
+        String presentableText = StringUtils.isNotEmpty(label) ? label : (insertText != null) ? insertText : "";
         String tailText = (detail != null) ? detail : "";
         LSPIconProvider iconProvider = GUIUtils.getIconProviderFor(wrapper.getServerDefinition());
         Icon icon = iconProvider.getCompletionIcon(kind);
@@ -881,12 +881,12 @@ public class EditorEventManager {
         String lookupString = null;
         if (textEdit != null) {
             lookupString = textEdit.getNewText();
-        } else if (!Strings.isNullOrEmpty(insertText)) {
+        } else if (StringUtils.isNotEmpty(insertText)) {
             lookupString = insertText;
-        } else if (!Strings.isNullOrEmpty(label)) {
+        } else if (StringUtils.isNotEmpty(label)) {
             lookupString = label;
         }
-        if (Strings.isNullOrEmpty(lookupString)) {
+        if (StringUtils.isEmpty(lookupString)) {
             return null;
         }
         // Fixes IDEA internal assertion failure in windows.
@@ -1011,7 +1011,7 @@ public class EditorEventManager {
                 String text = edit.getNewText();
                 Range range = edit.getRange();
 
-                if (range != null && !Strings.isNullOrEmpty(text)) {
+                if (range != null && StringUtils.isNotEmpty(text)) {
                     int start = DocumentUtils.LSPPosToOffset(editor, range.getStart());
                     int end = DocumentUtils.LSPPosToOffset(editor, range.getEnd());
                     lspEdits.add(new LSPTextEdit(text, start, end));
@@ -1025,7 +1025,7 @@ public class EditorEventManager {
                 String text = edit.getText();
                 int start = edit.getStartOffset();
                 int end = edit.getEndOffset();
-                if (Strings.isNullOrEmpty(text)) {
+                if (StringUtils.isEmpty(text)) {
                     document.deleteString(start, end);
                 } else {
                     text = text.replace(DocumentUtils.WIN_SEPARATOR, DocumentUtils.LINUX_SEPARATOR);

